### DISCUSSION TITLE
Upload to Postman completed with status: 400 - Bad Request

### DISCUSSION THREAD

**Discussion by PaulCluver** [apideck-libraries/portman#578] [Q&A] [answered]
Hi there Portman community, 

Hopefully someone can help me here. I am trying to automatically sync my generated postman collection to postman and I keep getting the following error: 

![image](https://github.com/apideck-libraries/portman/assets/5680263/300d06d0-7112-4631-bcb7-d810b3602c45)

Debugging this a bit deeper, the following error is being returned from the PostmanAPIService: 

![image](https://github.com/apideck-libraries/portman/assets/5680263/b22657f9-de63-4c5a-99d8-4df5bd0c0ad6)

**Comment by thim81** [comment]
hi @PaulCluver 

To not overload the Postman API when syncing the generated Postman collections, Portman keeps a local cache of the workspaces and collections in the workspace.

It might be that there is something off in the cache which causes the error to occur.

Some CLI options to try:
```
--postmanRefreshCache      Postman sync will refresh all local cached Postman API data (default: false)     [boolean]
--postmanFastSync          Postman sync creates new collection (new UID),instead of update (default: false) [boolean]
```

If none, of this works, could you try to find the file .portman.cache.json (usually in the "tmp" folder), and remove that file?
Portman will try to rebuild the cache on the next sync towards Postman.

**Reply by PaulCluver** [reply]
Hi Tim, 

Thanks so much for the advice. I will give it a whirl and revert back. 

Kind regards,
Paul.

**Comment by PaulCluver** [comment]
Hi Tim,

Unfortunately, none of the above suggestions worked. I will keep digging and update this post when I have found the root cause of the 400 response. 

Kind regards, 
Paul

**Reply by thim81** [reply]
Can you try to remove the collection from the Postman app?

**Reply by PaulCluver** [reply]
I have tried to remove the collection. The strange thing is that I can manually import it without any exceptions.

**Reply by thim81** [reply]
Sometimes the postman API has it quirks. 

Some more options to try out:
- verify if the API key is still valid
- Remove the collection from Postman, be sure to remove it from the postman trash since it is soft-deleted. Next remove the Portman cache file

**Reply by PaulCluver** [reply, marked as answer]
Hi Tim, 

After a bit of debugging a bit, I decided to extract the values being sent to the below method: 

![image](https://github.com/apideck-libraries/portman/assets/5680263/c752edf7-e7c1-4a8b-9e52-c8d8c974ee0c)

I then went into my Postman instance and tried to update the collection using the portman generated postman collection. This revealed a malformedRequestError as detailed below: 

![image](https://github.com/apideck-libraries/portman/assets/5680263/29b7d305-aade-4918-b43b-178a5fcee7a0)

I do not believe this to be an issue with Portman. Rather, it is related to the API spec file which is being built and maintained in Mulesoft. I fixed this by simply adding the following to the API spec before sending the file to Portman. 

![image](https://github.com/apideck-libraries/portman/assets/5680263/9320b320-d6d9-414c-8d8b-d63e33406723)

Kind regards,
Paul

**Reply by thim81** [reply]
hi @PaulCluver 

We only recently introduced the inclusion of the Postman info.version data, which in your case did not properly got convert.
So that might be a bug.

Could you share what your OpenAPI info looks like?
Does it have a version?
<img width="1041" alt="image" src="https://github.com/apideck-libraries/portman/assets/952446/bcc567c9-74ac-44df-b16d-9072318ea349">

**Reply by PaulCluver** [reply]
Hi Tim, 

Please find below the version information: 

![image](https://github.com/apideck-libraries/portman/assets/5680263/162529ab-e279-4035-b088-a6e58a6b85df)

Regards,
Paul

**Reply by thim81** [reply]
That is interesting, we will add a test case and try to improve the version conversion to handle the swagger2 versioning.

**Reply by thim81** [reply]
hi @PaulCluver 

We have a [PR](https://github.com/apideck-libraries/portman/pull/583) ready that should solve the "version":"v1" and other version related cases.

**Reply by PaulCluver** [reply]
This is great news. Thanks Tim

**Reply by thim81** [reply]
hi @PaulCluver 

We just released Portman v1.27.0, which includes the improvements for the version conversion.
It should solve your reported issue.

Hope you can use the new release and validate that the issue is resolved.

Thanks for your input and clear feedback, which helped us to improve Portman.
We always appreciate feedback and suggestions, so don't hesitate to share them.

**Reply by PaulCluver** [reply]
Thanks Tim, 

I will give it a try now.

**Reply by PaulCluver** [reply]
Hi Tim, 

I have upgraded to Portman v1.27.0. 

I am happy to report that the aforementioned issue is no longer happening. I can sync my collection to postman without getting a 400 response due to a malformed request body. 

Kind regards,
Paul.

**Reply by thim81** [reply]
Thanks for the confirmation
